plugins/mac-capture/window-utils.m

```diff
@@ -37,186 +36,205 @@
 #define WAIT_TIME_US WAIT_TIME_MS * 1000
 #define WAIT_TIME_NS WAIT_TIME_US * 1000
 
-bool find_window(cocoa_window_t cw, obs_data_t *settings, bool force)
+bool find_window(cocoa_window_t cw, obs_data_t *settings)
 {
-	if (!force && cw->next_search_time > os_gettime_ns())
+	uint64_t ts = os_gettime_ns();
+
+	if (cw->last_search_time + WAIT_TIME_NS > ts)
 		return false;
 
-	cw->next_search_time = os_gettime_ns() + WAIT_TIME_NS;
+	cw->last_search_time = ts;
 
-	pthread_mutex_lock(&cw->name_lock);
+	if (pthread_mutex_lock(&cw->mutex))
+		abort();
 
 	if (!cw->window_name.length && !cw->owner_name.length)
-		goto invalid_name;
+		goto unlock;
 
 	for (NSDictionary *dict in enumerate_windows()) {
```

Comment:
See previous PR's unresolved thread about this line.  Have all concerns been addressed?

@MakersF

https://github.com/obsproject/obs-studio/pull/4863#discussion_r724591985